packages/components/Badge/src/CounterBadge/CounterBadge.tsx

```diff
@@ -27,16 +40,15 @@ export const CounterBadge = compose<CounterBadgeType>({
   useRender: (userProps: CounterBadgeProps, useSlots: UseSlots<CounterBadgeType>) => {
     const iconProps = createIconProps(userProps.icon);
     const badge = useCounterBadge(userProps);
-
-    const Slots = useSlots(badge.props, (layer) => badgeLookup(layer, badge.props));
+    const Slots = useSlots(badge.props, (layer) => counterBadgeLookup(layer, badge.props));
 
     return (final: CounterBadgeProps, ...children: ReactNode[]) => {
       const { count, icon, iconPosition = 'before', overflowCount, dot, ...mergedProps } = mergeProps(badge.props, final);
       const { showBadge } = badge.state;
       const displayCount = count && count > overflowCount ? `${overflowCount}+` : `${count}`;
       const hasChildren = Children.toArray(children)[0];
 
-      return showBadge ? (
+      return showBadge || hasChildren ? (
```

Comment:
Seems like this component is meant to work when there are children as seen by line 57. Why is there a behavior change?

Reply:
"showBadge" is only true when count is passed or dot is passed.
So,
`<CounterBadge>Badge</CounterBadge>` will not display anything.
 `<CounterBadge count={10}>Badge</CounterBadge>` but this will display a counterBadge with just the text "Badge". So count is an unnecessary requirement here. Hence the additional check to display counterBadge even if just a child is passed